public/apps/configuration/panels/role-view/role-view.tsx

```diff
@@ -93,6 +97,9 @@ export function RoleView(props: RoleViewProps) {
   const [roleTenantPermission, setRoleTenantPermission] = useState<RoleTenantPermissionView[]>([]);
   const [toasts, addToast, removeToast] = useToastState();
   const [isReserved, setIsReserved] = useState(false);
+  const [isDeleteConfirmModalVisible, setIsDeleteConfirmModalVisible] = useState(false);
+  const closeDeleteConfirmModal = () => setIsDeleteConfirmModalVisible(false);
+  const showDeleteConfirmModal = () => setIsDeleteConfirmModalVisible(true);
```

Comment:
Saw there is also a delete confirm modal on tenant list. Please refactor to a utility function (like what we did for toast) so in the future it would be easy if we want to apply to deletion on all pages.

Reply:
Refactored in role-view. Will generate another PR for tenant list.